server/sonar-search/src/main/java/org/sonar/search/SearchServer.java

```diff
@@ -50,56 +57,177 @@ public SearchServer(Props props) {
 
   @Override
   public void start() {
-    Jmx.register(EsSettingsMBean.OBJECT_NAME, settings);
-    initBootstrap();
-    Settings esSettings = settings.build();
-    if (esSettings.getAsInt(MIMINUM_MASTER_NODES, 1) >= 2) {
-      LOGGER.info("Elasticsearch is waiting {} for {} node(s) to be up to start.",
-        esSettings.get(INITIAL_STATE_TIMEOUT),
-        esSettings.get(MIMINUM_MASTER_NODES)
-      );
+    Path absolutePath = findExecutable();
+
+    List<String> command = new ArrayList<>();
+    command.add(absolutePath.toString());
+    Map<String, String> settingsMap = settings.build();
+    settingsMap.entrySet().stream()
+      .filter(entry -> !"path.home".equals(entry.getKey()))
+      .forEach(entry -> command.add("-E" + entry.getKey() + "=" + entry.getValue()));
+    url = "http://"+settingsMap.get("http.host") + ":" + settingsMap.get("http.port");
+    System.out.println(command.stream().collect(Collectors.joining(" ")));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)